File: src/HasCompletedUnitRule.java
public class HasCompletedUnitRule implements RequirementRule {
    private Unit unit;

    public HasCompletedUnitRule(Unit unit) {
        this.unit = unit;
    }

    @Override
    public boolean isSatisfied(Student student) {
        return student.hasCompleted(unit);
    }
}

File: src/has_completed_unit_rule.cpp
#include "has_completed_unit_rule.hpp"

#include "student.hpp"
#include "unit.hpp"

namespace course_planner {

HasCompletedUnitRule::HasCompletedUnitRule(const Unit& unit) : unit_(&unit) {}

bool HasCompletedUnitRule::is_satisfied(const Student& student) const {
    return student.has_completed(*unit_);
}

}  // namespace course_planner
